docs/implementation-details.md

```diff
@@ -0,0 +1,71 @@
+TODO: Introduce the document
+
+### Structure
+The data structure of the sortition pool is a root node, which is a single
+`uint256`, followed by 6 layers of branch nodes, followed by a leaf layer. Each
+branch node is a `uint256`, and the branch structure is implemented as a
+`mapping(uint256 => mapping(uint256 => uint256))`. The first index represents
+the layer, and the second index represents the position within the layer.
+
+The first branch layer has 8 branches, and every subsequent layer has 8x the number of branches as the previous layer, so 
+
+1) node layer: 1
+2) branch layer 1: 8
+3) branch layer 2: 64
+4) branch layer 3: 512
+5) branch layer 4: 4,096
+6) branch layer 5: 32,768
+7) branch layer 6: 262,144
+8) leaf layer: 2,097,152
+
+### Leaf Serialization and Deserialization
+
+```
+function make(
+  address _operator,
+  uint256 _creationBlock,
+  uint256 _id
+) internal pure returns (uint256) {
+  assert(_creationBlock <= type(uint64).max);
+  assert(_id <= type(uint32).max);
+  uint256 op = uint256(bytes32(bytes20(_operator)));
+  uint256 uid = _id & ID_MAX;
+  uint256 cb = (_creationBlock & BLOCKHEIGHT_MAX) << ID_WIDTH;
+  return (op | cb | uid);
+}
+```
+
+The `Leaf.make` takes in an operator address, the block it was created in, and
+a monotonically increasing id for unique operators. We convert the operator
```

Comment:
Yeah, the first unique operator gets a 0, then the second gets a 1, then the third gets a 2, and so on. The assignment never decreases and never stays constant. If an operator leaves and comes back, they get their original ID back (since they're not unique).